### ISSUE TITLE
broken `@docType package` documentation

### ISSUE THREAD

**Issue by bastistician** [r-lib/roxygen2#1491]
**roxygen2** 7.0.0 (2019-11-12) had the breaking change

> * Using `@docType package` no longer automatically adds a `-package` alias.
>   Instead document `_PACKAGE` to get all the defaults for package documentation

from <https://github.com/r-lib/roxygen2/commit/29db38fb823fcfe9cf2ad05f885bf20dfa1c964b>.

However, thousands of packages still use the old way ([that was still recommended a year ago](http://web.archive.org/web/20220607010527/https://r-pkgs.org/man.html#man-packages))
```
#' @docType package
#' @name pkgname
NULL
```
not being aware that they should

- either document the new ["special sentinel" `_PACKAGE`](https://r-pkgs.org/man.html#sec-man-package-doc)
- or manually add `@aliases <pkgname>-package`

because they no longer automatically get that alias that is required for a package overview page to be found via `package?pkgname`. As a consequence, CRAN now has thousands of packages where this help request no longer works.

Could **roxygen2** warn users when there is a `@docType package` block without corresponding `@aliases` (and possibly recommend the new `_PACKAGE` feature)?

**Comment by bastistician** [comment]
Addendum: I noticed that some packages do use the `"_PACKAGE"` syntax introduced long ago via @krlmlr's #392, but explicitly remove the default aliases via `@aliases NULL` (e.g., <https://github.com/krlmlr/enc/commit/3ee7849381ae8b8177fd85912e487cfaea5ff05b>) producing a help page without alias, unaccessible from dynamic `help()`.

I think it would be very useful if **roxygen2** actively warned its users about the breaking change from 7.0.0, i.e., when the old approach `@docType package` on `NULL` is used (without manually specifying `@alias pkgname-package`) and recommend documenting `"_PACKAGE"`. Maybe @krlmlr would have an idea?

**Comment by MichaelChirico** [comment]
Just came to post the same -- the current warning is misleading, routing users to the more awkward `#' @name pkgname-package` approach:

```
[pkgname-package.R:10] Block must have a @name
ℹ Either document an existing object or manually specify with @name
```

The `"_PACKAGE"` sentinel approach strikes me as a bit redundant (we've already written `#' @docType package`), but if this is the requirement we should be steering users there more helpfully.

**Comment by hadley** [comment]
Definitely open to this, although I'm not sure if it's still worthwhile given that CRAN already told everyone about the probelm 😞

**Comment by MichaelChirico** [comment]
There are still a ton of packages on CRAN with the issue:

https://github.com/search?q=lang%3AR+%2F%40docType%5Cspackage%2F+%2F%5ENULL%24%2F+org%3Acran&type=code

**Comment by hadley** [comment]
Good point, definitely worth thinking about what the right fix is here.

**Comment by hadley** [comment]
Change was made in https://github.com/r-lib/roxygen2/commit/29db38fb823fcfe9cf2ad05f885bf20dfa1c964b as part of some other refactoring with comment " `@docType package` no longer adds `-package` alias as this broke encapsulation and was only mildly useful."

**Comment by hadley** [comment]
I'd appreciate someone would take a look at #1535 and confirm that the approach makes sense — the basic is to make `@docType package` essentially an alias for `_PACKAGE` (with deprecation message), ensuring that the behaviour of the two can no longer drift out of sync.

**Comment by zachmayer** [comment]
This is a really, really dumb question, but how do I make a `_PACKAGE` document?  Is this just like a new file I add to package folder?  Is a `_PACKAGE` something else?

**Comment by MichaelChirico** [comment]
@zachmayer maybe a useful reference example 

https://github.com/tidyverse/tibble/blob/main/R/tibble-package.R

**Comment by zachmayer** [comment]
lol I was overthinking it.  As always a reference was all I needed.   Thank you!

https://github.com/tidyverse/tibble/blob/main/R/tibble-package.R#L62

**Comment by bbolker** [comment]
**tl;dr** can anyone explain why I'm getting a spurious `\alias{-package}` in my `.Rd` file?

Confused on this issue, don't know if anyone can help me out. I'm maintaining a legacy package. The roxygen stuff at the head of the package-defining .R file looks like this:

```
#' @rdname margins
#' @name margins
#' @aliases margins-package
#' @docType package
#' @title Marginal Effects Estimation
```

But the head of `margins.R` looks like this:

```
\docType{package}
\name{margins}
\alias{-package}
\alias{margins}
\alias{margins-package}
```

This is with roxygen2 7.3.2.  I *could* rearrange everything to follow the now-recommend `_PACKAGES` approach, but if I can minimally fix what I'm doing wrong I would prefer it.

[Here's the repo](https://github.com/bbolker/margins) if anyone wants to look at the original.
